Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -664,15 +651,71 @@ void ReportAttack(wstring basename, wstring charname, uint system)
 	return;
 }
 
+// For all players in the base's system, resync their reps towards all objects
+// of this base.
+void PlayerBase::SiegeModChainReaction(uint client)
+{
+	map<uint, PlayerBase*>::iterator it;
+	for (it = player_bases.begin(); it != player_bases.end(); it++)
+	{
+		if (it->second->system == this->system)
+		{
+			if (HkDistance3D(it->second->position, this->position) < siege_mod_chain_reaction_trigger_distance)
+			{
+				if (!(it->second->siege_mode))
+				{
+					float attitude = it->second->GetAttitudeTowardsClient(client, true);
+					if (attitude < -0.55f)
+					{
+						it->second->siege_mode = true;
+
+						const wstring& charname = (const wchar_t*)Players.GetActiveCharacterName(client);
+						ReportAttack(it->second->basename, charname, it->second->system, L"has detected hostile activity at the nearby base by");
+
+						it->second->SyncReputationForBase();
+					}
+				}
+			}
+		}
+	}
+}
+
+// For all players in the base's system, resync their reps towards this object.
+void PlayerBase::SyncReputationForBaseObject(uint space_obj)
+{
+	struct PlayerData *pd = 0;
+	while (pd = Players.traverse_active(pd))
+	{
+		if (pd->iShipID && pd->iSystemID == system)
+		{
+			int player_rep;
+			pub::SpaceObj::GetRep(pd->iShipID, player_rep);
+			float attitude = GetAttitudeTowardsClient(pd->iOnlineID);
+
+			int obj_rep;
+			pub::SpaceObj::GetRep(space_obj, obj_rep);
+			pub::Reputation::SetAttitude(obj_rep, player_rep, attitude);
+		}
+	}
+}
+
 // Return true if 
 float PlayerBase::SpaceObjDamaged(uint space_obj, uint attacking_space_obj, float curr_hitpoints, float new_hitpoints)
 {
+	float incoming_damage = abs(curr_hitpoints - new_hitpoints);
+
 	// Make sure that the attacking player is hostile.
 	uint client = HkGetClientIDByShip(attacking_space_obj);
 	if (client)
 	{
 		const wstring &charname = (const wchar_t*)Players.GetActiveCharacterName(client);
 		last_attacker = charname;
+
+		hostile_tags_damage[charname] += incoming_damage;
+
+		if (hostile_tags_damage.find(charname) == hostile_tags_damage.end())
```

Comment:
shouldn't this if block come before the line above? right now it looks like dead code

Reply:
Huh. Indeed.